src/_nebari/provider/cicd/github.py

```diff
@@ -143,49 +143,34 @@ class GHA_on_extras(BaseModel):
     paths: List[str]
 
 
-class GHA_on(BaseModel):
-    # to allow for dynamic key names
-    __root__: Dict[str, GHA_on_extras]
-
-    # TODO: validate __root__ values
-    # `push`, `pull_request`, etc.
-
-
-class GHA_job_steps_extras(BaseModel):
-    # to allow for dynamic key names
-    __root__: Union[str, float, int]
+GHA_on = RootModel[Dict[str, GHA_on_extras]]
+GHA_job_steps_extras = RootModel[Union[str, float, int]]
```

Comment:
Need to read more at https://docs.pydantic.dev/1.10/usage/models/#custom-root-types